### ISSUE TITLE
Failed missing split

### ISSUE THREAD

**Issue by mghotz** [LuigiVampa92/xapk-to-apk#9]
I tried this script over 4 xapk. Same error for all.
adb: failed to install xxx-aligned-debugSigned.apk: Failure [INSTALL_FAILED_MISSING_SPLIT: Missing split for package_name]

**Comment by cristan** [comment]
I have the same thing for https://apkcombo.com/maersk/com.maersk.trackandtrace.maerskline/download/apk (version 8.0.6 to be exact).

**Comment by cristan** [comment]
Ah, https://github.com/makesey/xapk-to-apk/tree/main works ^_^. So it would be cool if https://github.com/LuigiVampa92/xapk-to-apk/pull/10 would get merged.

Not that it works for the Maersk APK, I then get another error. I now get:
> INSTALL_FAILED_INVALID_APK: Failed to extract native libraries, res=-2

**Comment by bubby963** [comment]
> Ah, https://github.com/makesey/xapk-to-apk/tree/main works ^_^. So it would be cool if #10 would get merged.
> 
> Not that it works for the Maersk APK, I then get another error. I now get:
> 
> > INSTALL_FAILED_INVALID_APK: Failed to extract native libraries, res=-2

Thanks a lot for pointing me in the right direction, this worked great!

As for your issue, I'm sure you've already solved it but in case you haven't, you need to decompile the APK using apktool or the like and change the value of the `extractNativeLibs` property within `AndroidManifest.xml` to `true`, then rebuild and sign the apk.

**Comment by cristan** [comment]
Thanks! I've now implemented setting `extractNativeLibs` to `true` in [my fork](https://github.com/cristan/xapk-to-apk). I think [the changes](https://github.com/LuigiVampa92/xapk-to-apk/compare/development...cristan:xapk-to-apk:development) look quite good so far.

Unfortunately, now after the holidays I don't get the same error anymore (even without this change), but I still get errors.

I still try to do with with the xapk of [this app](https://apkcombo.app/maersk/com.maersk.trackandtrace.maerskline/download/apk) and my xapktoapk.sign.properties looks like this:

```
sign.enabled=true
sign.keystore.file=/Users/cristan/.android/debug.keystore
sign.keystore.password=android
sign.key.alias=androiddebugkey
sign.key.password=android
```

And the error I get:
```
The APK failed to install.
Error: INSTALL_PARSE_FAILED_MANIFEST_MALFORMED: Failed parse during installPackageLI: /data/app/vmdl2029447969.tmp/base.apk (at Binary XML file line #72): requires an android:value or android:resource attribute
```

Output of logcat:
```
01-08 16:40:32.384   368   408 W PackageParser: Unknown element under <manifest>: queries at /data/app/vmdl1672987882.tmp/base.apk Binary XML file line #9
01-08 16:40:32.386   368   408 W PackageManager: Failed parse during installPackageLI: /data/app/vmdl1672987882.tmp/base.apk (at Binary XML file line #72): <meta-data> requires an android:value or android:resource attribute
```

I don't get that at all. All <meta-data> files have an android:value or android:resource attribute and a queries element in the <manifest> element is quite normal.

**Comment by bubby963** [comment]
Interesting, could I get you to decompile the resultant APK using APK tool like so

`$apktool d <outputfile.apk>`

and send the contents of line 72 of `AndroidManifest.xml`?

I have mine working except I'm using this for vulnerability testing of android apps, so on top of running xapk-to-apk.py I then decompile the package, make some changes to `network_security_config.xml` and `AndroidManifest.xml`, then recompile and sign it. Doing this I'm able to install the apps fine but sometimes it requires some manual tuning based on the app.

**Comment by cristan** [comment]
Line 72:

```
<meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@null"/>
```

It looks innocuous, but this is definitely the problem. If I add this to my own app, I indeed can't install it.

This is coming from apktool: when unzip the xapk and then decompile the main APK, I also get this line in the AndroidManifest.xml. Maybe this is a bug in `apktool`? I've also checked whether any other apks have the `default_notification_icon`, but that's not the case. I wonder why this is happening.

Anyway, thank you random person on the internet ❤️ When I remove this line, I can indeed install the APK 😃.

**Comment by bubby963** [comment]
Yup I had the exact same issue just yesterday so I thought that was it haha. The `@null` seems to be a non-valid resource name causing it to fail, but it is added by apktool for some reason. Might be an issue there.

Anyway glad you got it working!

**Comment by cristan** [comment]
Ah, as two people has had this same issue now, I've now created #13 for this.

**Comment by LuigiVampa92** [comment]
Hi, just checked how things work after merging your pull requests. I have managed to build the apk and install it, but I getting stuck on the first screen. Though, it is true for the apk that I have downloaded and installed as well. Probably the app just doesn't want to run on an emulator, but it successfully installed.

Can you please also check and ensure that the problem is fixed, and if that is so, close this issue. Thank you!
